docs/understand/hipgraph.rst

```diff
@@ -0,0 +1,81 @@
+    :doc:`HIP graph API documentation<../doxygen/html/group___graph>`.
+
+Graph format
+================================================================================
+
+A HIP graph is, like any other graph, made up of nodes and edges. The nodes of a
+HIP graph represent the operations performed, while the edges mark dependencies
+between those operations.
+
+The nodes can consist of:
+
+- empty nodes
+- nested graphs
+- kernel launches
+- host-side function calls
+- HIP memory functions (copy, memset, ...)
+- HIP events
+- signalling or waiting on external semaphores
+
+The following figure visualizes the concept of graphs, compared to using streams.
+
+.. figure:: ../data/understand/hipgraph/hip_graph.svg
+    :alt: Diagram depicting the difference between using streams to execute
+          kernels with dependencies, resolved by explicitly calling
+          hipDeviceSynchronize, or using graphs, where the edges denote the
+          dependencies.
+
+HIP graph advantages
+================================================================================
+
+The standard way of launching work on GPUs via streams incurs a small overhead for each operation involved
+every time. For kernels that take a considerable amount to finish, this overhead
+usually is negligible, however many workloads, including scientific simulations
+and AI, involve launching many relatively small kernels repeatedly for many iterations.
+
+HIP graphs have been specifically designed to tackle this problem by only
+requiring one launch from the host per iteration, and minimizing that overhead
+by performing most of the initialization beforehand. Graphs may provide
+additional performance benefits, by enabling optimizations that are only
+possible when knowing the dependencies between the operations.
+
+HIP graph usage
+================================================================================
+
+Using HIP graphs to execute your work requires three different steps, where the
+first two are the initial setup and only need to be executed once. First the
+definition of the operations (nodes) and the dependencies (edges) between them.
+The second step is the instantiation of the graph. This takes care of validating
+and initializing the graph, to reduce the overhead when executing the graph.
+
+The third step is the actual execution of the graph, which then takes care of
+launching all the kernels and executing the operations while respecting their
+dependencies and necessary synchronizations as specified.
+
+As HIP graphs require some set up and initialization overhead before their first
+execution, they only provide a benefit for workloads that require many iterations to complete.
+
+Setting up HIP graphs
+================================================================================
+
+HIP graphs can be created by explicitly defining them, or using stream capture.
+For the available functions see the
+:doc:`HIP graph API documentation<../doxygen/html/group___graph>`.
```

Comment:
The ending seems a little sudden.  I think an example graph would be great to have here.

Reply:
What about the figure earlier in this document comparing launching kernels via streams and the graph next to it? I think it fits better up there.

Also the goal would be to have a how-to page, that would be linked to here. However that section is not finished, as I don't have the means to test the code currently. Also we needed to reconsider how to handle this, as the graph API is still in beta.